graylog2-web-interface/src/components/sidecars/configuration-forms/ConfigurationVariablesHelper.jsx

```diff
@@ -0,0 +1,151 @@
+            this.errorModal.open();
+          // Not in use, ask for confirmation
+          } else {
+            deleteConfirmationModal.open();
+          }
+        });
+      };
+
+      const _handleDeleteConfirm = () => {
+        ConfigurationVariableActions.delete(configVar)
+          .then(() => this._onSuccessfulUpdate());
+      };
+
+      const escapedName = `\${${configVar.name}}`;
+      variableRows.push(
+        <tr key={this._getId(configVar.id)}>
+          <td><code>{escapedName}</code></td>
+          <td>{configVar.description}</td>
+          <td style={{ textAlign: 'right' }}>
+            <BootstrapModalConfirm ref={(c) => { deleteConfirmationModal = c; }}
+                                   title="Delete Configuration Variable?"
+                                   onConfirm={_handleDeleteConfirm}
+                                   onCancel={() => {}}>
+              <p>Are you sure you want to remove the configuration variable <strong>{configVar.name}</strong>?</p>
+            </BootstrapModalConfirm>
+            <Button bsStyle="primary" bsSize="xsmall" onClick={_handleDeleteCheck}>
+              Delete
+            </Button>
+            <EditConfigurationVariableModal id={configVar.id}
+                                            name={configVar.name}
+                                            description={configVar.description}
+                                            content={configVar.content}
+                                            create={false}
+                                            saveConfigurationVariable={this._saveConfigurationVariable} />
+          </td>
+        </tr>,
+      );
+    });
+    return variableRows;
+  };
+
+  _isLoading = () => {
+    return !(this.state.configurationVariables);
+  };
+
+  _saveConfigurationVariable = (configurationVariable, callback) => {
+    ConfigurationVariableActions.save.triggerPromise(configurationVariable)
+      .then(() => this._onSuccessfulUpdate(callback));
+  };
+
+  _onSuccessfulUpdate = (callback) => {
+    if (typeof callback === 'function') {
+      callback();
+    }
+    this._reloadConfiguration();
+  };
+
+  render() {
+    if (this._isLoading()) {
+      return <Spinner />;
+    }
+
+    return (
+      <div>
+        <EditConfigurationVariableModal create
+                                        saveConfigurationVariable={this._saveConfigurationVariable} />
+        <br />
+        <br />
+        <br />
+        <div className={`table-responsive ${ConfigurationHelperStyle.tableMaxHeight}`}>
+          <Table responsive>
+            <thead>
+              <tr>
+                <th>Name</th>
+                <th>Description</th>
+                <th style={{ textAlign: 'right' }} >Actions</th>
+              </tr>
+            </thead>
+            <tbody>
+              {this._configurationVariableListFormatter()}
+              <BootstrapModalWrapper ref={(modal) => { this.errorModal = modal; }}>
```

Comment:
I would move the modal out of the table to make the markup a bit cleaner. In the end react-bootstrap takes care of picking the container for the modal, so it does not need to be in here.